server.js

```diff
@@ -86,14 +101,23 @@ wsServer.on('connection', function connection(ws, request) {
                 case 'CreateRoom':
                     console.log('CreateRoom Request received from client %o', jsonObj)
                     roomId = util.createRoom(lobby, jsonObj.playerId, jsonObj.playerName, jsonObj.playerAvatar, jsonObj.maxPlayersPerRoom, jsonObj.playerTags)
+                    if (roomId !== undefined) {
+                        wsStatsServer.emit('updateStats')
+                    }
                     break
                 case 'JoinRoom':
                     console.log('JoinRoom Request received from client %o', jsonObj)
                     roomId = util.joinRoom(lobby, jsonObj.roomId, jsonObj.playerId, jsonObj.playerName, jsonObj.playerAvatar, jsonObj.playerTags)
+                    if (roomId !== undefined) {
+                        wsStatsServer.emit('updateStats')
+                    }
                     break
                 case 'ExitRoom':
                     console.log('ExitRoom Request received from client %o', jsonObj)
-                    roomId = util.exitRoom(lobby, jsonObj.roomId, jsonObj.playerId)
+                    exitRoomSuccess = util.exitRoom(lobby, jsonObj.roomId, jsonObj.playerId)
+                    if (exitRoomSuccess === true) {
+                        wsStatsServer.emit('updateStats')
```

Comment:
One thing that I think we probably want to do is aggregate some of these stats in memory for a short amount of time (10 seconds?) before pushing to your Websockets server. I'm thinking that if EdgeMultiplay gets super popular in the future, all of these emits directly to your server could overwhelm it. 

Also getLobbyStats has to recalculate all of the stats everytime an event happens, so if you aggregate these as you go, your server will have more bandwidth to do non-stats related things.

Reply:
Thanks for the suggestion, I agree will add those stats aggregation in the future.